File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C/s081988820_comments_removed.c
#include <stdio.h>
#include <stdlib.h>

#define MAX 1024

typedef struct {
    int H, W;
    int c[10][10];
    int min[10];
    int ans;
} Main;

void solve(Main *main) {
    int H, W;
    scanf("%d %d", &H, &W);
    main->H = H;
    main->W = W;

    for (int i = 0; i < 10; i++) {
        for (int j = 0; j < 10; j++) {
            scanf("%d", &main->c[i][j]);
        }
    }

    for (int i = 0; i < 10; i++) {
        main->min[i] = main->c[i][1];
    }

    for (int tc = 0; tc < 10; tc++) {
        for (int i = 0; i < 10; i++) {
            for (int j = 0; j < 10; j++) {
                if (main->min[i] > main->c[i][j] + main->min[j]) {
                    main->min[i] = main->c[i][j] + main->min[j];
                }
            }
        }
    }

    main->ans = 0;
    for (int h = 1; h <= H; h++) {
        for (int w = 1; w <= W; w++) {
            int A;
            scanf("%d", &A);
            if (A >= 0) {
                main->ans += main->min[A];
            }
        }
    }

    printf("%d\n", main->ans);
}

int main() {
    Main main;
    solve(&main);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Python/s081988820_comments_removed.py
import sys

class Main:
    def __init__(self):
        pass

    def solve(self):
        import sys
        input = sys.stdin.read
        data = input().split()
        index = 0

        H = int(data[index])
        W = int(data[index + 1])
        index += 2

        c = [[0] * 10 for _ in range(10)]
        for i in range(10):
            for j in range(10):
                c[i][j] = int(data[index])
                index += 1

        min_cost = [c[i][1] for i in range(10)]
        for _ in range(10):
            for i in range(10):
                for j in range(10):
                    min_cost[i] = min(min_cost[i], c[i][j] + min_cost[j])

        ans = 0
        for h in range(1, H + 1):  # Adjusted to 1-based indexing
            for w in range(1, W + 1):  # Adjusted to 1-based indexing
                A = int(data[index])
                index += 1
                if A >= 0:
                    ans += min_cost[A]

        print(ans)

if __name__ == "__main__":
    main = Main()
    main.solve()
# End of Code
